nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/pgp/PGPControllerService.java

```diff
@@ -0,0 +1,893 @@
+     */
+    private Collection<ValidationResult> validateForDecrypt(PropertyContext context) {
+        PGPPrivateKey key = getPrivateKey(context);
+        if (key != null) {
+            return Collections.emptySet();
+        }
+
+        char[] passphrase = getPBEPassPhrase(context);
+        if (passphrase != null) {
+            return Collections.emptySet();
+        }
+
+        final List<ValidationResult> problems = new ArrayList<>();
+        problems.add(new ValidationResult.Builder()
+                .subject(CONTROLLER_NAME)
+                .valid(false)
+                .explanation("Controller requires a secret key or PBE pass-phrase for Decrypt operations.")
+                .build());
+        return problems;
+    }
+
+
+    /**
+     * This method validates a {@link PropertyContext} for signing.
+     *
+     * @param context describing signing keys
+     * @return validation results, which may be empty if given context is valid
+     */
+    private Collection<ValidationResult> validateForSign(PropertyContext context) {
+        PGPPrivateKey key = getPrivateKey(context);
+        if (key != null) {
+            return Collections.emptySet();
+        }
+
+        final List<ValidationResult> problems = new ArrayList<>();
+        problems.add(new ValidationResult.Builder()
+                .subject(CONTROLLER_NAME)
+                .valid(false)
+                .explanation("Controller requires a secret key for Sign operations.")
+                .build());
+        return problems;
+    }
+
+
+    /**
+     * This method validates a {@link PropertyContext} for verifying.
+     *
+     * @param context describing verify keys
+     * @return validation results, which may be empty if given context is valid
+     */
+    private Collection<ValidationResult> validateForVerify(PropertyContext context) {
+        PGPPublicKey key = getPublicKey(context);
+        if (key != null && key.isEncryptionKey()) {
+            return Collections.emptySet();
+        }
+
+        final List<ValidationResult> problems = new ArrayList<>();
+        problems.add(new ValidationResult.Builder()
+                .subject(CONTROLLER_NAME)
+                .valid(false)
+                .explanation("Controller requires a public key for Verify operations.")
+                .build());
+        return problems;
+    }
+
+
+    private PGPPublicKey getPublicKey(PropertyContext context) {
+        PGPPublicKeys keys = null;
+
+        if (context.getProperty(PUBLIC_KEYRING_TEXT).isSet()) {
+            final String keyText = context.getProperty(PUBLIC_KEYRING_TEXT).getValue();
+            keys = readPublicKeys(keyText);
+
+        } else if (context.getProperty(PUBLIC_KEYRING_FILE).isSet()) {
+            final String keyFile = context.getProperty(PUBLIC_KEYRING_FILE).getValue();
+            try {
+                keys = readPublicKeys(new File(keyFile));
+            } catch (FileNotFoundException ignored) {
```

Comment:
Why is this exception ignored? I know we validate the property descriptor, but I think there can be a race condition here where the file is moved/removed from disk between validation and extraction.

Reply:
Good catch, that was supposed to have a return.  So updated.